### ISSUE TITLE
Lose focus when Alt key pressed (Ubuntu 14.x)

### ISSUE THREAD

**Issue by gsemet** [Guake/guake#352]
When the Alt key is pressed, when the "hide window on focus loss" is set, the window is closed, preventing the use of "Alt+..." hotkeys

**Comment by enzochiau** [comment]
For the gnome-terminal on Ubuntu, it can use `ALT` key as `Emacs` binding. For example `ALT+b` is backward a word and `ALT+f` is forward a word, etc.

So I am not sure what do you mean `the system stole` the hotkey.

**Comment by falconepl** [comment]
As @enzochiau said, I believe there could be a workaround for that. If you have Gedit opened and you hit `Alt`, "type command" prompt shows up... but when you have e.g. GNOME Terminal opened and you hit `Alt`, the prompt does *not* show up.

**Comment by falconepl** [comment]
For anyone interested... If you're OK with disabling HUD pop-up on `Alt` key press, you need to disable both "Key to show the HUD when tapped" and "Key to show the menu bar while pressed" in *Ubuntu Unity Plugin* settings in *CompizConfig settings manager* (or replace it with other shortcut than `Alt`).

**Comment by Stibbons** [comment]
indeed you have the right answer!

**Comment by falconepl** [comment]
Yeah, you're right. That's out of the Guake's scope to prevent being hidden while window manager is configured with default settings.

**Comment by onelaview** [comment]
It seems even after turning off "Key to show HUD" and "Key to show menu bar" hotkey via CompizConfig Manager, hotkey like `Alt`+`Shift`+something is still not applicable in Guake. Immediately after I hit `Alt`+`Shift`, Guake is going disappeared.

**Comment by brandonsturgeon** [comment]
This is super inconvenient! I need to copy text from Tmux, but I have to use Alt to do that, but if I use Alt then my Guake window disappears becuase I need to have "Hide on lose focus" on, because if I accidentally Alt-Tab (happens when I don't realize my VM is focused and I try to alt-tab in my host machine) , your Guake window is gone forever.

**Comment by Anticom** [comment]
@onelaview unchecking "Hide on lost focus" fixed the issue for me. (Also using 0.8.4)

**Comment by brandonsturgeon** [comment]
It's worth noting that while "Hide on lost focus" does fix this issue, it causes a lot of inconveniences down the road, especially if you use a VM.

If this bug were fixed (I don't even know what's wrong, it might be infeasible) then we wouldn't have to resort to repetitively annoying workarounds.

**Comment by gsemet** [comment]
sadly... but if one is couragous enough, Pull Request are welcomed!

**Comment by brandonsturgeon** [comment]
Haha, does that imply that it's possible to make a fix on this end? If not, where would the fix/change have to be made? On ubuntu's end?

**Comment by CurvesTech** [comment]
For me In the settings, there is an option 'Hide on Lose Focus' it was checked, Unchecking it gave me my desired behaviour. I open and close guake with F12, and don't really rely on 'close on lose focus feature'.

**Comment by brandonsturgeon** [comment]
It sucks because that setting is actually pretty useful, but it's a good solution atm.

**Comment by gsemet** [comment]
still reproduced.

**Comment by aichingm** [comment]
This also happens with kde and the Super-Key which opens the menu (kickoff).

**Comment by eodus** [comment]
There is a solution for Ubuntu (tested on 14 and 18, probably works for other versions). One can completely disable Alt shortcut using CCSM:

1. Install compizconfig-settings-manager
> sudo apt-get install compizconfig-settings-manager
2. Run CCSM
> ccsm
3. Select Desktop -> Ubuntu Unity Plugin 
4. Disable "Key to show menu bar while pressed" AND "Key to show the HUD when tapped"

**Comment by gsemet** [comment]
would you accept to put this remark in the documentation ?
